### PR TITLE
Add the func of getting system and metrics snapshot by pid

### PR DESCRIPTION
**Pull request by douzl**
Signed-off-by: Zhongqiang Dou <douzl0814@163.com>

### CODE REVIEW

**Comment by andygrunwald** [comment]
Thank you for this.
Right now i am on vacation and i will be back in ~9 days. Then i will have
a look.
Just to let you know :)

Am Samstag, 10. September 2016 schrieb Coveralls :

File: snapshot.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+)
+
+// GetMetricsSnapshot will return the snapshot of pid
+func (c *Client) GetMetricsSnapshot(pid *Pid) (map[string]interface{}, error) {
```
Comment (andygrunwald) [inline snapshot.go]: Would it make sense to add a new struct for the snapshot endpoint like you did for System?
With this we get rid of the interface{} and the necessary casting afterwards.
You applied the dummy json already in tests/master1.metrics.snapshot.json

**Comment by andygrunwald** [comment]
Thank you @douzl.
It looks fine :) I made a comment related to the map[string]interface{}.

File: snapshot.go (hunk 1 above)
Comment (douzl) [inline snapshot.go]: The fields are too much. I will handle it later. :)

**Comment by douzl** [comment]
@andygrunwald Updated with the struct MetricsSnapshot in type.go， please help to review. Thanks!

**Comment by andygrunwald** [comment]
Looks good, thank you @douzl!

**Pull request merged**